idaes/models_extra/power_generation/unit_models/soc_submodels/solid_oxide_cell.py

```diff
@@ -0,0 +1,994 @@
+import pyomo.environ as pyo
+from pyomo.network import Port
+
+from idaes.core import declare_process_block_class, UnitModelBlockData
+from idaes.models.unit_models.heat_exchanger import HeatExchangerFlowPattern
+import idaes.models_extra.power_generation.unit_models.soc_submodels as soc
+import idaes.models_extra.power_generation.unit_models.soc_submodels.common as common
+from idaes.models_extra.power_generation.unit_models.soc_submodels.common import (
+    _constF,
+    _species_list,
+    _element_list,
+    _element_dict,
+)
+import idaes.core.util.scaling as iscale
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.solvers import get_solver
+
+import idaes.logger as idaeslog
+
+
+@declare_process_block_class("SolidOxideCell")
+class SolidOxideCellData(UnitModelBlockData):
+    CONFIG = UnitModelBlockData.CONFIG()
+    CONFIG.declare(
+        "control_volume_zfaces",
+        ConfigValue(
+            description="List containing coordinates of control volume faces "
+            "in z direction. Coordinates must start with zero, be strictly "
+            "increasing, and end with one"
+        ),
+    )
+    CONFIG.declare(
+        "control_volume_xfaces_fuel_electrode",
+        ConfigValue(
+            description="List containing coordinates of control volume faces "
+            "in x direction for fuel electrode. Coordinates must start with "
+            "zero, be strictly increasing, and end with one"
+        ),
+    )
+    CONFIG.declare(
+        "control_volume_xfaces_oxygen_electrode",
+        ConfigValue(
+            description="List containing coordinates of control volume faces "
+            "in x direction for oxygen electrode. Coordinates must start with "
+            "zero, be strictly increasing, and end with one"
+        ),
+    )
+    CONFIG.declare(
+        "control_volume_xfaces_electrolyte",
+        ConfigValue(
+            description="List containing coordinates of control volume faces "
+            "in x direction for electrolyte. Coordinates must start with "
+            "zero, be strictly increasing, and end with one"
+        ),
+    )
+    CONFIG.declare(
+        "fuel_component_list",
+        ConfigValue(
+            default=["H2", "H2O"], description="List of components in fuel stream"
+        ),
+    )
+    CONFIG.declare(
+        "oxygen_component_list",
+        ConfigValue(
+            default=["O2"], description="List of components in the oxygen stream"
+        ),
+    )
+    CONFIG.declare(
+        "fuel_triple_phase_boundary_stoich_dict",
+        ConfigValue(
+            default={"H2": -0.5, "H2O": 0.5, "e^-": 1.0},
+            description="Dictionary with species as keys and stoichiometric coefficients as values "
+            "for the redox reaction that occurs at the fuel-side triple phase boundary",
+        ),
+    )
+    CONFIG.declare(
+        "inert_fuel_species_triple_phase_boundary",
+        ConfigValue(
+            default=[],
```

Comment:
I think this might cause unexpected problems, see https://github.com/IDAES/idaes-pse/issues/330.

I think it would be better to make the `None`, and then later check for this and handle appropriately.

Reply:
Changing the default to `None` is easy enough, but I just realized that we would have the same issue anywhere we include a `dict` or `list` as a default config argument, empty or not. I looked at the implementation of `pyomo.common.config.ConfigBase` and it doesn't appear to create a copy when a default value is given. I think I don't ever mutate these lists or dictionaries in any submodule, so it wouldn't be a major problem that they all share the same default list or dictionary, but that might not be true in all IDAES or Pyomo models. 

Just making every `list`/`dict` default argument `None`, then filling in actual default values later would work, but it would be a lot more confusing to the user than using `None` to represent an empty `list`/`dict`. @jsiirola , any thoughts?